creator/releases/models.py

```diff
@@ -0,0 +1,429 @@
+    )
+    version = VersionField(
+        partial=False,
+        coerce=False,
+        default=next_version,
+        help_text="Semantic version of the release",
+    )
+    is_major = models.BooleanField(
+        default=False,
+        help_text="Whether the release is a major version change or not",
+    )
+    created_at = models.DateTimeField(
+        auto_now_add=True, help_text="Date created"
+    )
+
+    @transition(field=state, source="waiting", target="initializing")
+    def initialize(self):
+        """ Begin initializing tasks """
+        return
+
+    @transition(field=state, source="initializing", target="running")
+    def start(self):
+        """ Start the release """
+        return
+
+    @transition(field=state, source="running", target="staged")
+    def staged(self):
+        """ The release has been staged """
+        return
+
+    @transition(field=state, source="staged", target="publishing")
+    def publish(self):
+        """ Start publishing the release """
+        return
+
+    @transition(field=state, source="publishing", target="published")
+    def complete(self):
+        """ Complete publishing """
+        if self.is_major:
+            self.version = self.version.next_major()
+        else:
+            self.version = self.version.next_minor()
+        self.save()
+        return
+
+    @transition(field=state, source=FAIL_SOURCES, target="canceling")
+    def cancel(self):
+        """ Cancel the release """
+        return
+
+    @transition(field=state, source="canceling", target="canceled")
+    def canceled(self):
+        """ The release has finished canceling """
+        return
+
+    @transition(field=state, source=FAIL_SOURCES, target="failed")
+    def failed(self):
+        """ The release failed """
+        return
+
+
+class ReleaseTask(models.Model):
+    """
+    A Release Task is a process that is run on a Task Service as part of a
+    Release.
+    """
+
+    class Meta:
+        permissions = [("list_all_releasetask", "Show all release tasks")]
+        get_latest_by = "created_at"
+
+    kf_id = models.CharField(max_length=11, primary_key=True, default=task_id)
+    uuid = models.UUIDField(
+        default=uuid.uuid4, help_text="UUID used internally"
+    )
+    state = FSMField(
+        default="waiting", help_text="The current state of the task"
+    )
+    progress = models.IntegerField(
+        default=0,
+        help_text="Optional field"
+        " representing what percentage of the task"
+        " has been completed",
+    )
+    release = models.ForeignKey(
+        Release,
+        on_delete=models.CASCADE,
+        null=False,
+        blank=False,
+        related_name="tasks",
+    )
+    release_service = models.ForeignKey(
+        "ReleaseService",
+        on_delete=models.CASCADE,
+        null=False,
+        blank=False,
+        related_name="tasks",
+    )
+    job_log = models.ForeignKey(
```

Comment:
This relation seems a little odd to me because its indirectly linking two things that maybe should be linked directly. I think of it like: 

- A Task is a job definition
- A Task is run to produce a Job
- A Job has a JobLog

So with this in mind, I would think you'd have a Task relate to a Job which relates to a JobLog?